### PR TITLE
Added Book entity, BookRepository and BookService

### PR DESCRIPTION
**Pull request by Serhii-Nosulia**

### CODE REVIEW

File: src/main/java/mate/bookshopapp/repository/BookRepository.java [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package mate.bookshopapp.repository;
+
+import mate.bookshopapp.model.Book;
+import org.springframework.data.jpa.repository.JpaRepository;
+
+public interface BookRepository extends JpaRepository<Book, Long> {
```
Comment (Ivan95kos) [inline src/main/java/mate/bookshopapp/repository/BookRepository.java]: You should not use `JpaRepository` in this part

`BookRepository` interface with methods and `BookRepositoryImpl` class:
```
Book save(Book book);
List findAll();
```

File: src/main/java/mate/bookshopapp/service/impl/BookServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+import mate.bookshopapp.repository.BookRepository;
+import mate.bookshopapp.service.BookService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
+public class BookServiceImpl implements BookService {
+    private final BookRepository bookRepository;
+
+    @Autowired
```
Comment (Ivan95kos) [inline src/main/java/mate/bookshopapp/service/impl/BookServiceImpl.java]: Here You don't need to use `@Autowired`, you can remove it

**Review by Ivan95kos — CHANGES_REQUESTED** [review]
Good job overall!

File: src/main/java/mate/bookshopapp/repository/BookRepository.java (hunk 1 above)
Comment (Serhii-Nosulia) [reply src/main/java/mate/bookshopapp/repository/BookRepository.java:6]: When I add SessionFactory to BookRepository, I have this problem, and I don't know how fix it
Знімок екрана 2023-11-10 о 20 04 43

**Review by Ivan95kos — APPROVED** [review]
Nice

**Pull request merged**